### PR TITLE
refactor bridge functionality into the vehicle plugin

### PR DESCRIPTION
**Pull request by codebot**
This PR refactors the Zenoh bridge functionality into the vehicle gateway plugin. This simplifies the usage of multirobot comms because it is then unnecessary to launch a separate bridge process, and it should be more efficient because it can re-use the already-existing comms to the autopilot of choice (e.g. the micro-ros agent talking to the PX4). To activate the zenoh multi-agent comms and state broadcasting, there is a new function, `create_multirobot_session()`, which expects to receive a filename to a JSON file which has a top-level `zenoh` object that can be extracted, dumped back to string, and then passed to Zenoh `zc_config_from_str()`.

The following steps will create a two-vehicle simulation and start streaming data between them:

First, create a two-vehicle simulation, with each vehicle (vehicle 1 and 2) operating fully in their own DDS domains (DDS domains 1 and 2, respectively):
```
export MULTIROBOT_CONFIG=~/vg/src/vehicle_gateway/px4_sim/config/two_isolated.yaml
ros2 launch px4_sim px4_sim_multi.launch.py
```

In a separate terminal, we can start the follower vehicle (2):
```
ROS_DOMAIN_ID=2 ros2 run vehicle_gateway_demo follower 2 ~/vg/src/vehicle_gateway/vehicle_gateway_demo/config/leader_follower_multicast_discovery.json
```

In a separate terminal, we can start the leader vehicle (1):
```
ROS_DOMAIN_ID=1 ros2 run vehicle_gateway_demo vtol_position_control  1 ~/vg/src/vehicle_gateway/vehicle_gateway_demo/config/leader_follower_multicast_discovery.json
```

At time of writing, the follower vehicle won't do anything other than print the leader's position to its console. The interesting part is that it is getting that position through Zenoh, since the DDS domains are isolated (to reflect a real-world scenario where each vehicle's DDS domain is fully contained onboard, to avoid discovery scaling issues).

Next up is to add logic to `vehicle_gateway_demo/src/follower.cpp` to do something more interesting than just print to console :smile:

### CODE REVIEW

File: vehicle_gateway_demo/src/follower.cpp [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  const double leader_north = j["north"].get<double>();
+  const double leader_east = j["east"].get<double>();
+  const double leader_down = j["down"].get<double>();
+  const double airspeed = j["airspeed"].get<double>();
+  printf(
+    "leader position: (%.3f, %.3f, %.3f)\n",
+    leader_north, leader_east, leader_down);
+
+  if (!this->following_active) {
+    this->start = std::chrono::steady_clock::now();
```
Comment (audrow) [inline vehicle_gateway_demo/src/follower.cpp:93]: Is `start` declared anywhere, or am I missing something?

Comment (audrow) [reply vehicle_gateway_demo/src/follower.cpp:93]: After removing this line, things built for me.

Comment (ahcorde) [reply vehicle_gateway_demo/src/follower.cpp]: https://github.com/osrf/vehicle_gateway/pull/122/commits/bdd2b71b3d3a46b8d52f9b080e542ffd5c4597cc

Comment (audrow) [reply vehicle_gateway_demo/src/follower.cpp]: I see it was removed in bdd2b71b3d3a46b8d52f9b080e542ffd5c4597cc

**Review by audrow — APPROVED** [review]
Looks good and works for me when I test it locally! Great demo!

**Pull request merged**